x-pack/plugin/apm-data/src/main/resources/ingest-pipelines/apm@default-pipeline.yaml

```diff
@@ -32,14 +32,25 @@ processors:
       (ctx.metricset?.name == 'service_destination')
     dataset: apm.service_destination.1m
 
-# Invoke user-defined custom pipelines, in ascending order
-# of specificity.
+# Invoke user-defined custom pipelines, in ascending order of specificity:
 - pipeline:
-    name: apm@custom
+    name: global@custom
     ignore_missing_pipeline: true
 - pipeline:
+    name: "{{{data_stream.type}}}@custom"
+    ignore_missing_pipeline: true
+- pipeline:
+    if: "ctx?.data_stream?.dataset != 'apm'"
```

Comment:
Nope, the null check below is because there's a method call. Here we can just rely on the `?.` operator, which will compare `null != 'apm'` and do the right thing in case there's no `data_stream.dataset` field.